### PR TITLE
Email lockdown

### PR DESCRIPTION
**Pull request by Pomax**
fixes #https://github.com/mozilla/network-pulse-api/issues/16

### CODE REVIEW

**Comment by Pomax** [comment]
Tested this with my `mozillafoundation.org` account, which works, and my `gmail.com` and `hotmail.com` accounts, which get blocked. Admin login not being a moz login is essentially irrelevant as it uses its own login portal on `/admin`

**Comment by alanmoo** [comment]
While this technically works, I'm concerned about the silent failure if non-Mozillians attempt to log in. I feel like we need some way to convey "you're not logged in but that's expected".

**Comment by Pomax** [comment]
fair enough - rather than a redirect to `/` we can redirect to `/login-failure` so that we can serve up some content with a 403?

**Comment by alanmoo** [comment]
Or maybe a query param so that users can still get to the same page when merged with #25?

**Comment by Pomax** [comment]
oh that's a good idea!

**Comment by Pomax** [comment]
Updated the PR so that it now does a callback with a `loggedin=...` that is either True or False depending on whether authentication succeeded or not, and will either redirect the user to `/` with that argument if no original_url is provided, or redirects to the original_url with that query pair added (with a toggle to make sure we use `?loggedin=...` or `&loggedin=...` depending on whether the original url already had query args)

File: pulseapi/entries/views.py [hunk 1]
```diff
@@ -21,8 +21,8 @@ def post_validate(request):
     nonce = False
 
     if request.data:
-        csrf_token = request.data['csrfmiddlewaretoken']
-        nonce = request.data['nonce']
+        csrf_token = request.data.get('csrfmiddlewaretoken', False)
```
Comment (Pomax) [inline pulseapi/entries/views.py]: yeah, it's a property of Python's dictionaries. You can either do direct access with array notation, or you can use `get` with a fallback, https://docs.python.org/2/library/stdtypes.html#dict.get

**Pull request merged**